File: Windows Forms/WindowsFormsApplication1/WindowsFormsApplication1/SQLiteWrapper.cs
using System.Collections.Generic;
using System.Data.SQLite;

namespace WindowsFormsApplication1
{
    public class SQLiteWrapper
    {
        private SQLiteConnection _connection = null;

        public SQLiteWrapper()
        {
        }

        public void CreateFile(string fileName)
        {
            SQLiteConnection.CreateFile(fileName);
        }

        public void OpenConnection(string connectionSettings)
        {
            _connection = new SQLiteConnection(connectionSettings);
            _connection.Open();
        }

        public string ExecuteNonQuery(string query)
        {
            SQLiteCommand command               = new SQLiteCommand(query, _connection);
            int affectedRows                    = command.ExecuteNonQuery();
            Dictionary<string, object> values   = new Dictionary<string, object>() { { "AffectedRows", affectedRows.ToString() } };
            string json                         = Json.JsonParser.ToJson(values);
            return json;
        }

        public string ExecuteSelect(string query)
        {
            SQLiteCommand command                 = new SQLiteCommand(query, _connection);
            SQLiteDataReader reader               = command.ExecuteReader(System.Data.CommandBehavior.Default);
            List<Dictionary<string, object>> rows = new List<Dictionary<string, object>>();

            while(reader.Read() && reader.HasRows)
            {
                Dictionary<string, object> currentRow = new Dictionary<string, object>();

                for(int i = 0; i < reader.FieldCount; i++)
                {
                    string key   = reader.GetName(i);
                    object value = reader.GetValue(i);

                    currentRow.Add(key, value);
                }

                rows.Add(currentRow);
            }

            Dictionary<string, object> values   = new Dictionary<string, object>() { { "rows", rows } };
            string json                         = Json.JsonParser.ToJson(values);

            return json;
        }
    }
}

File: Android/ConceptTest/app/src/main/java/com/test/leo/concepttest/SQLiteWrapper.java
package com.test.leo.concepttest;

import android.content.pm.PackageManager;
import android.database.sqlite.*;
import android.database.*;

import org.json.JSONArray;
import org.json.JSONObject;

import java.io.File;
import java.util.*;

/**
 * Created by Leo on 6/25/2016.
 */
public class SQLiteWrapper
{
    private MainActivity _activity = null;
    private SQLiteDatabase _database = null;
    private File _file = null;

    public SQLiteWrapper(MainActivity activity)
    {
        _activity = activity;
    }

    public void CreateFile(String fileName)
    {
        _file = new File(android.os.Environment.getExternalStorageDirectory(), "/MyDatabase.sqlite");

        if(!_file.exists())
        {
            try
            {
                String[] perms = {"android.permission.READ_EXTERNAL_STORAGE", "android.permission.WRITE_EXTERNAL_STORAGE"};
                int permsRequestCode = 200;
                _activity.requestPermissions(perms, permsRequestCode);
                boolean hasPermissions = _activity.checkSelfPermission("android.permission.WRITE_EXTERNAL_STORAGE") == PackageManager.PERMISSION_GRANTED;

                if(hasPermissions)
                    _file.createNewFile();
            }
            catch(Exception e)
            {
                String message = e.getMessage();
                _activity.ShowDialog("Exception", message);
            }
        }
    }

    public void OpenConnection(String connectionSettings)
    {
        if(_file.exists())
            _database = SQLiteDatabase.openOrCreateDatabase(_file, null);
        //else
            //_activity.ShowDialog("Error", "Database file does not exist");
    }

    public String ExecuteNonQuery(String query)
    {
        /*Cursor cursor = _database.rawQuery(query, null);
        ArrayList<HashMap<String, String>> rows = new ArrayList<HashMap<String, String>>();

        while(cursor.moveToNext())
        {
            HashMap<String, String> currentRow = new HashMap<String, String>();

            for(int i = 0; i < cursor.getColumnCount(); i++)
            {
                String key   = cursor.getColumnName(i);
                String value = cursor.getString(i);

                currentRow.put(key, value);
            }

            rows.add(currentRow);
        }

        HashMap<String, ArrayList> result = new HashMap<String, ArrayList>();
        result.put("rows", rows);
        JSONObject jsonObject = new JSONObject(result);

        //_activity.ShowDialog("Result", jsonObject.toString());

        return jsonObject.toString();*/

        _database.execSQL(query);

        long affectedRows               = GetAffectedRows();
        HashMap<String, String> result  = new HashMap<String, String>();
        result.put("affected_rows", Long.toString(affectedRows));
        JSONObject jsonObject = new JSONObject(result);
        return jsonObject.toString();
    }

    private long GetAffectedRows()
    {
        SQLiteStatement statement = _database.compileStatement("SELECT changes()");
        long affectedRows         = statement.simpleQueryForLong();

        return affectedRows;
    }

    public String ExecuteSelect(String query)
    {
        Cursor cursor = _database.rawQuery(query, null);
        ArrayList<HashMap<String, String>> rows = new ArrayList<HashMap<String, String>>();

        while(cursor.moveToNext())
        {
            HashMap<String, String> currentRow = new HashMap<String, String>();

            for(int i = 0; i < cursor.getColumnCount(); i++)
            {
                String key   = cursor.getColumnName(i);
                String value = cursor.getString(i);

                currentRow.put(key, value);
            }

            rows.add(currentRow);
        }

        HashMap<String, JSONArray> result = new HashMap<String, JSONArray>();
        result.put("rows", new JSONArray(rows));
        JSONObject jsonObject = new JSONObject(result);

        return jsonObject.toString();
    }
}
